src/hooks/use-extended-theme/use-extended-theme.ts

```diff
@@ -0,0 +1,62 @@
+import { stardust, useMemo, useTheme } from '@nebula.js/stardust';
+import type { ExtendedTheme, BackgroundColors } from './types';
+import isTransparentColor from '../../utils/is-transparent-color';
+import isDarkColor from '../../utils/is-dark-color';
+
+type ChartBackgroundResolver = (theme: stardust.Theme) => string | undefined;
+
+/**
+ * The colors in the chart can depend on background colors set on the chart, object and sheet level.
+ * Even though not officially supported, it is expected that css settings on object and sheet are respected.
+ * The priority order is: chart theme > object css > object theme
+ * If the result for object/chart background is transparent, the sheet css color is used
+ */
+export const getBackgroundColors = (
+  theme: ExtendedTheme,
+  rootElement: HTMLElement,
+  chartBackgroundResolver?: ChartBackgroundResolver,
+  objectBackgroundResolver?: ChartBackgroundResolver
+): BackgroundColors => {
+  const qvInnerObject = rootElement?.closest('.qv-object .qv-inner-object');
+  const objectColorFromCSS = qvInnerObject && window.getComputedStyle(qvInnerObject).backgroundColor;
+
+  const qvPanelSheet = rootElement?.closest('.qv-panel-sheet');
+  const sheetColorFromCSS = qvPanelSheet && window.getComputedStyle(qvPanelSheet).backgroundColor;
+
+  // chartBackgroundFromTheme is undefined if nothing is set specifically for the chart (ex: object.straightTableV2.backgroundColor)
+  const chartBackgroundFromTheme = chartBackgroundResolver?.(theme);
```

Comment:
A "background color" does not really exist in the `sn-pivot-table`. So I made these resolver functions optional.